src/components/savingsAccounts/IRA.tsx

```diff
@@ -0,0 +1,281 @@
+  // fields about contributions from form 5498
+  contributions: data.contributions.toString(),
+  rolloverContributions: data.rolloverContributions.toString(),
+  rothIraConversion: data.rothIraConversion.toString(),
+  recharacterizedContributions: data.recharacterizedContributions.toString(),
+  requiredMinimumDistributions: data.requiredMinimumDistributions.toString(),
+  lateContributions: data.lateContributions.toString(),
+  repayments: data.repayments.toString()
+})
+
+export default function IRA(): ReactElement {
+  const ira = useYearSelector(
+    (state: TaxesState) => state.information.individualRetirementArrangements
+  )
+
+  const people: Person[] = useYearSelector((state: TaxesState) => [
+    state.information.taxPayer?.primaryPerson,
+    state.information.taxPayer?.spouse
+  ])
+    .filter((p) => p !== undefined)
+    .map((p) => p as Person)
+
+  const dispatch = useYearDispatch()
+
+  const methods = useForm<IraUserInput>()
+  const { handleSubmit } = methods
+
+  const { navButtons, onAdvance } = usePager()
+
+  const onSubmitAdd = (formData: IraUserInput): void => {
+    dispatch(addIRA(toIra(formData)))
+  }
+
+  const onSubmitEdit =
+    (index: number) =>
+    (formData: IraUserInput): void => {
+      dispatch(editIRA({ index, value: toIra(formData) }))
+    }
+
+  const hsaBlock = (
+    <FormListContainer<IraUserInput>
+      items={ira.map((a) => toIraUserInput(a))}
+      onSubmitAdd={onSubmitAdd}
+      onSubmitEdit={onSubmitEdit}
+      removeItem={(i) => dispatch(removeIRA(i))}
+      icon={() => <Work />}
+      primary={(ira: IraUserInput) => ira.payer}
+      secondary={(ira: IraUserInput) => (
+        <span>
+          {IraPlanTypeTexts[ira.planType]}
+          <br />
+          gross distribution: <Currency value={toIra(ira).grossDistribution} />
+          <br />
+          contribution: <Currency value={toIra(ira).contributions} />
+          <br />
+        </span>
+      )}
+    >
+      <Grid container spacing={2}>
+        <LabeledInput
+          name="payer"
+          label="payer for this account"
+          patternConfig={Patterns.plain}
+          sizes={{ xs: 12, lg: 6 }}
+        />
+        <GenericLabeledDropdown<IraPlanType>
+          label="IRA Type"
+          dropDownData={Object.values(IraPlanType)}
+          valueMapping={(x) => x}
+          keyMapping={(_, i) => i}
+          textMapping={(status) => IraPlanTypeTexts[status]}
+          name="planType"
+        />
+        <GenericLabeledDropdown
+          dropDownData={people}
+          label="Recipient"
+          valueMapping={(p: Person, i: number) =>
+            [PersonRole.PRIMARY, PersonRole.SPOUSE][i]
+          }
+          name="personRole"
+          keyMapping={(p: Person, i: number) => i}
+          textMapping={(p: Person) =>
+            `${p.firstName} ${p.lastName} (${formatSSID(p.ssid)})`
+          }
+        />
+        <LabeledInput
+          name="contributions"
+          label="Contributions"
```

Comment:
Check out [`boxLabel`](https://github.com/ustaxes/UsTaxes/blob/323fa4716c8ae24e85a8e21052425eb17c92ee1c/src/components/income/W2JobInfo.tsx#L230), it'd be nice to explicitly map these to box numbers from the 1099 or 5498